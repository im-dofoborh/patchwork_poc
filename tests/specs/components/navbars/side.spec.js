import { mount, createLocalVue } from '@vue/test-utils';
import VueRouter from 'vue-router';
import Vuex from 'vuex';
import SideNavbar from '~/components/navbars/Side';

describe('SideNavbar component', () => {
  let localVue = null;
  let wrapper = null;
  let router = null;
  let store = null;

  beforeEach(() => {
    localVue = createLocalVue();
    localVue.component('side-navbar', SideNavbar);
    localVue.use(VueRouter);
    router = new VueRouter();
    localVue.use(Vuex);
    store = new Vuex.Store({
      state: {
        componentsLoaded: false,
        components: [],
      },
    });
    wrapper = mount(SideNavbar, {
      localVue,
      router,
      store,
    });
  })

  it('renders default variant', () => {
    expect(wrapper.html()).toMatchSnapshot();
  });
});
