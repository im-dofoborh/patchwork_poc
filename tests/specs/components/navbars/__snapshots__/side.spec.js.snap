// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SideNavbar component renders default variant 1`] = `
<nav class="sg-side-nav">
  <section class="hidden-md-up">
    <div class="sg-side-nav__menu">
      <!---->
      <ul class="sg-side-nav__list">
        <li class="sg-side-nav__item "><a href="#/overview" class="sg-side-nav__link">
            Overview
          </a></li>
        <li class="sg-side-nav__item "><a href="#/foundations" class="sg-side-nav__link">
            Structural Foundations
          </a></li>
        <li class="sg-side-nav__item "><a href="#/components" class="sg-side-nav__link">
            Component Library
          </a></li>
        <li class="sg-side-nav__item "><a href="#/language" class="sg-side-nav__link">
            Visual Language
          </a></li>
      </ul>
      <div class="sg-side-nav__divider"></div>
    </div>
  </section>
  <!---->
</nav>
`;
